src/components/homepage/UpEvents.astro

```diff
@@ -0,0 +1,89 @@
+---
+
+interface EventData {
+  id: string;
+  Title: string;
+  // would be better to have a description field
+  By: string[];
+  Date: string;
+  Time: string;
+  URL: string;
+  IMGURL: string; // this is the url to the section on html page so it cannot be used for the image
+  BannerURL: string; // there is no banner url but should be added for latest entries to the events.json
+  ProfileLink: string[];
+}
+
+let eventsData: EventData[];
+try {
+  const response = await fetch('https://raw.githubusercontent.com/noskofficial/noskofficial.github.io/main/events.json');
```

Comment:
@axyut how about using relative path instead of url fetching
something like

https://github.com/noskofficial/ncitalums/blob/main/src/utils/loadconfig.ts

Reply:
will work, but won't you need to maintain two files now? instead of just one in the official repo?

Reply:
Yeah both @mukezhz  Daii and you are on line @axyut . I think he is mentioning about url for the events on the config file. So, that someone who may use this template later on can just change the url for the events on the config file.

In our case, lets put the url to the events.json of the nosk website repo as value for eventsUrl on config file.

Reply:
yeah I got it, forgot about it being a template

Reply:
> And, for the image url lets keep on default if its not configured on the events details.

what should be the default image ? should I leave what is now as it is?